### PR TITLE
Taught PulpImport to not rely on next_version from upstream.

### PR DESCRIPTION
**Pull request by ggainey**
Also taught import and export tests to share more utility code.

fixes #8116

### CODE REVIEW

File: CHANGES/8116.bugfix [hunk 1]
```diff
@@ -0,0 +1,4 @@
+Taught ''PulpImport'' correct way to find and import ''RepositoryVersions''.
+
+Previous implementation only worked for ''RepositoryVersions'' that were
+the 'current' version of the exported ''Repository''.
```
Comment (daviddavis) [inline CHANGES/8116.bugfix:4]: use double backticks for inline literals. Also, I'd probably just make this one paragraph.

File: pulpcore/tests/functional/api/using_plugin/test_pulpimport.py [hunk 2]
```diff
@@ -421,3 +423,50 @@ def test_import_check_multiple_errors(self):
...
+        # export from version-0 to version-1, last=v1
+        body = {
+            "start_versions": [versions.results[0].pulp_href],
+            "versions": [versions.results[1].pulp_href],
+            "full": False,
+        }
+        export = self._gen_export(exporter, body)
+        return export
+
+    def test_import_not_latest_version(self):
```
Comment (daviddavis) [inline pulpcore/tests/functional/api/using_plugin/test_pulpimport.py:454]: :1st_place_medal:

**Review by daviddavis — APPROVED** [review]
One small comment. Other than that, looks good to me.

File: CHANGES/8116.bugfix (hunk 1 above)
Comment (ggainey) [reply CHANGES/8116.bugfix]: Well that's a fun typo - good catch!

Comment (ggainey) [reply CHANGES/8116.bugfix]: RE one paragraph - huh. I think of tc-news same as I do commits - headline, blank, body. Is the 'right' formatting written down in our docs somewhere?  Anyway - noted, change made.

File: pulpcore/tests/functional/api/using_plugin/test_pulpimport.py (hunk 2 above)
Comment (ggainey) [reply pulpcore/tests/functional/api/using_plugin/test_pulpimport.py:454]: :)

**Review by goosemania — COMMENTED** [review]
Great job and moar tests! Please check one comment about `rv_name`.

File: pulpcore/app/tasks/importer.py [hunk 3]
```diff
@@ -145,12 +138,19 @@ def import_repository_version(importer_pk, destination_repo_pk, source_repo_name
                 )
             )
 
+        rv_path = ""
         # Extract the repo version files
         with tarfile.open(tar_path, "r:gz") as tar:
             for mem in tar.getmembers():
-                if re.match(fr"^{_repo_version_path(src_repo)}/.+", mem.name):
+                match = re.search(fr"(^repository-{source_repo_name}_[0-9]+)/.+", mem.name)
```
[the remark this answers is not in the document]
Comment (ggainey) [reply pulpcore/app/tasks/importer.py:145]: It's a regex, 'fr' means either "freakin' ridiculous" or just "ridiculous" (the 'f' is often silent in acronyms...)

File: pulpcore/app/tasks/importer.py [hunk 4]
```diff
@@ -145,12 +138,19 @@ def import_repository_version(importer_pk, destination_repo_pk, source_repo_name
...
+                if match:
+                    rv_name = match.group(1)
                     tar.extract(mem, path=temp_dir)
 
+        if not rv_name:
+            raise ValidationError("No RepositoryVersion found for {}".format(rv_name))
```
[the remark this answers is not in the document]
Comment (ggainey) [reply pulpcore/app/tasks/importer.py:151]: We should I18N anything that can be seen by users. We're def inconsistent, but that's no excuse to not do the right thing. Will fix.

File: pulpcore/app/tasks/importer.py [hunk 5]
```diff
@@ -145,12 +138,19 @@ def import_repository_version(importer_pk, destination_repo_pk, source_repo_name
                 )
             )
 
+        rv_path = ""
```
[the remark this answers is not in the document]
Comment (ggainey) [reply pulpcore/app/tasks/importer.py:141]: ....sonofa....

Outstanding catch, thank you. And none of my testing trips this condition. Will fix.

[the remark this answers is not in the document]
Comment (goosemania) [reply pulpcore/app/tasks/importer.py:141]: Sweet, thanks!

**Review by goosemania — APPROVED** [review]

**Pull request merged**